packages/workflow-core/src/lib/utils/context-transformers/types.ts

```diff
@@ -0,0 +1,18 @@
+import { TContext } from '../types';
+
+export type TTransformationLogic = string;
+export abstract class BaseContextTransformer {
+  abstract name: string;
+  type = 'context-transformer';
+  context: TContext;
+  transformationLogic: TTransformationLogic;
+  options: {};
+
+  protected constructor(context: TContext, transformationLogic: TTransformationLogic, options: {}) {
```

Comment:
1. it looks like an interface with a constructor,
2. that's a bad practice to have a constructor in the base class as one of the things which separates  implementations is the constructor itself
3. also it makes the code less readable as you need to go to the base class every time you wanna understand the child implementation

Reply:
yeah inheritance has its issues.

but for now, it is built as intendant. maybe later we'll have to change the constructor.
later on, if the abstract constructor will require additional functionalities/requirements by any other "transformer". we'll update the logic. for now it works as intendant